io-sim/src/Control/Monad/IOSim/Internal.hs

```diff
@@ -372,25 +417,25 @@ schedule !thread@Thread{
           !expiry  = d `addTime` time
           !timers' = snd (PSQ.alter updateTimeout_ tmid timers)
           !thread' = thread { threadControl = ThreadControl k ctl }
-      trace <- schedule thread' simstate { timers = timers' }
+      !trace <- schedule thread' simstate { timers = timers' }
       return (SimTrace time tid tlbl (EventTimerUpdated tmid expiry) trace)
 
     -- updating a negative timer is a no-op, unlike in `GHC.Event`.
     UpdateTimeout (NegativeTimeout _tmid) _d k ->
       {-# SCC "schedule.UpdateTimeout" #-} do
-      let thread' = thread { threadControl = ThreadControl k ctl }
+      let !thread' = thread { threadControl = ThreadControl k ctl }
       schedule thread' simstate
 
-    CancelTimeout (Timeout tvar _tvar' tmid) k ->
+    CancelTimeout (Timeout tvar tmid) k ->
       {-# SCC "schedule.CancelTimeout" #-} do
       let !timers' = PSQ.delete tmid timers
           !thread' = thread { threadControl = ThreadControl k ctl }
       !written <- execAtomically' (runSTM $ writeTVar tvar TimeoutCancelled)
-      (!wakeup, wokeby) <- threadsUnblockedByWrites written
-      mapM_ (\(SomeTVar var) -> unblockAllThreadsFromTVar var) written
+      (wakeup, wokeby) <- threadsUnblockedByWrites written
+      !_ <- mapM_ (\(SomeTVar var) -> unblockAllThreadsFromTVar var) written
```

Comment:
Is this necessary? Again, if so, separate patch.

Reply:
Done